src/main/java/frc/robot/subsystems/shooter/ShooterIOTalonFX.java

```diff
@@ -0,0 +1,373 @@
+package frc.robot.subsystems.shooter;
+
+import static frc.robot.subsystems.shooter.ShooterConstants.*;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.MotionMagicConfigs;
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicExpoVoltage;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.AbsoluteSensorRangeValue;
+import com.ctre.phoenix6.signals.FeedbackSensorSourceValue;
+import com.ctre.phoenix6.signals.GravityTypeValue;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import com.ctre.phoenix6.signals.SensorDirectionValue;
+import edu.wpi.first.math.util.Units;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team3061.util.ArmSystemSim;
+import frc.lib.team3061.util.VelocitySystemSim;
+import frc.lib.team6328.util.TunableNumber;
+
+public class ShooterIOTalonFX implements ShooterIO {
+
+  // Using VelocityTorqueCurrentFOC to set the velocity of the motors
+  private VelocityTorqueCurrentFOC shootMotorTopVelocityRequest;
+  private VelocityTorqueCurrentFOC shootMotorBottomVelocityRequest;
+  private MotionMagicExpoVoltage angleMotorPositionRequest;
+
+  // Using StatusSignal to get the stator current of the motors
+  private StatusSignal<Double> shootMotorTopStatorCurrentStatusSignal;
+  private StatusSignal<Double> shootMotorBottomStatorCurrentStatusSignal;
+  private StatusSignal<Double> angleMotorStatorCurrentStatusSignal;
+
+  // Using StatusSignal to get the supply current of the motors
+  private StatusSignal<Double> shootMotorTopSupplyCurrentStatusSignal;
+  private StatusSignal<Double> shootMotorBottomSupplyCurrentStatusSignal;
+  private StatusSignal<Double> angleMotorSupplyCurrentStatusSignal;
+
+  // Using StatusSignal to get the velocity of the motors
+  private StatusSignal<Double> shootMotorTopVelocityStatusSignal;
+  private StatusSignal<Double> shootMotorBottomVelocityStatusSignal;
+  private StatusSignal<Double> angleMotorPositionStatusSignal;
+
+  private StatusSignal<Double> shootMotorTopReferenceVelocityStatusSignal;
+  private StatusSignal<Double> shootMotorBottomReferenceVelocityStatusSignal;
+  private StatusSignal<Double> angleMotorReferencePositionStatusSignal;
+
+  private StatusSignal<Double> shootMotorTopTemperatureStatusSignal;
+  private StatusSignal<Double> shootMotorBottomTemperatureStatusSignal;
+  private StatusSignal<Double> angleMotorTemperatureStatusSignal;
+
```

Comment:
add status signals for the top and bottom motors closed loop reference slope which can be obtained via the getClosedLoopReferenceSlope method and corresponds to the Motion Magic target velocity.